src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -2790,6 +2804,7 @@ internal static double GetGenSizeBeforeMB(List<TraceGC> GCs, TraceGC gc, Gens ge
                 return heapStats.GenerationSize2 / 1000000.0;
             }
 
+            // TODO, poh, what shall we do here?
```

Comment:
Note - this is important.

Reply:
Good question. I would defer to @maoni0 on this.

Reply:
oops, I missed this comment. is there a problem doing the same as the other gens here, ie, return heapStats.GenerationSize4?

Reply:
I am probably just being paranoid. Notice the `Math.Max` special case there for LOH just a few lines below. Wonder why we do it and does it apply to POH or not.

Reply:
the Max just says that we want to the bigger one because we want the size before GC - either we do a gen2 GC, and this gc's heapstats could have a smaller number, in which if the previous one's is bigger we know that's closer to the actual value; or if it's not a gen2, and the currrent GC's is bigger then something was allocated and increased the gen size so we should take this gc's data. this is all an estimate. hopefully we never hit this code path because the perheap hist already tells us the accurate size_before so this is only there to handle the unusual cases. but the point is really there's no reason to handle LOH and POH differently.